src/main/java/net/sacredlabyrinth/phaed/dynmap/simpleclans/DynmapSimpleClans.java

```diff
@@ -1,181 +1,196 @@
+        } catch (IllegalStateException ex) {
+            getLogger().severe(ex.getMessage());
+            return false;
+        }
+
+        saveDefaultImages();
+        loadTasks();
+        loadLayers();
+
+        return true;
+    }
+
+    public void loadLayers() {
+        ConfigurationSection clanHomesSection = Objects.requireNonNull(getConfig().getConfigurationSection("layer.homes"));
+        ConfigurationSection killsSection = Objects.requireNonNull(getConfig().getConfigurationSection("layer.kills"));
+        ConfigurationSection landsSection = Objects.requireNonNull(getConfig().getConfigurationSection("layer.lands"));
+
+        String defaultHomeIcon = clanHomesSection.getString("default-icon", DefaultIcons.CLANHOME.getName());
+
+        IconStorage homesIcons = new IconStorage(this, "/images/clanhome", defaultHomeIcon, markerApi);
+        IconStorage killsIcons = new IconStorage(this, "/images", DefaultIcons.BLOOD.getName(), markerApi);
+
+        try {
+            homesLayer = new HomesLayer(getClanManager(), homesIcons, new LayerConfig(clanHomesSection), markerApi);
+        } catch (IllegalStateException ex) {
+            debug(ex.getMessage());
+        }
+
+        try {
+            killsLayer = new KillsLayer(killsIcons, new LayerConfig(killsSection), markerApi);
+        } catch (IllegalStateException ex) {
+            debug(ex.getMessage());
+        }
+
+        try {
+            // Running on next ticks to safely receive lands coordinates
+            getScheduler().runTask(this, () ->
+                    landsLayer = new LandsLayer(getClanManager(), simpleclans.getProtectionManager(), new LayerConfig(landsSection), markerApi));
+        } catch (IllegalStateException ex) {
+            debug(ex.getMessage());
+        }
+    }
+
+    @NotNull
+    public ClanManager getClanManager() {
+        return simpleclans.getClanManager();
+    }
+
+    @NotNull
+    public DynmapAPI getDynmapApi() {
+        return dynmapApi;
+    }
+
+    @Nullable
+    public HomesLayer getHomeLayer() {
+        return homesLayer;
+    }
+
+    @Nullable
+    public KillsLayer getKillsLayer() {
+        return killsLayer;
+    }
+
+    @Nullable
+    public LandsLayer getLandsLayer() {
+        return landsLayer;
+    }
+
+    private void loadDependencies() {
+        dynmapApi = (DynmapAPI) getPluginManager().getPlugin("DynMap");
+        simpleclans = (SimpleClans) getPluginManager().getPlugin("SimpleClans");
+
+        if (simpleclans == null || dynmapApi == null) {
+            getPluginLoader().disablePlugin(this);
+        }
+
+        if (simpleclans == null) {
+            throw new IllegalStateException("SimpleClans wasn't found, disabling...");
+        }
+
+        if (dynmapApi == null) {
+            throw new IllegalStateException("Dynmap wasn't found, disabling...");
+        }
+        markerApi = dynmapApi.getMarkerAPI();
+        if (markerApi == null) {
+            getPluginLoader().disablePlugin(this);
```

Comment:
There are two calls to disablePlugin().
Both could be moved to the catch in loadDependencies().